File: yangxj96-common/src/main/java/com/yangxj96/saas/common/base/BaseService.java
package com.yangxj96.saas.common.base;

import com.baomidou.mybatisplus.extension.service.IService;

/**
 * RESTFul 接口公用service层
 *
 * @param <O> 子类对应的实体
 *            </O>
 */
public interface BaseService<O extends BaseEntity> extends IService<O> {

    O create(O datum);

    Boolean delete(String id);

    O modify(O datum);

}

File: yangxj96-common/src/main/kotlin/com/yangxj96/saas/common/base/BaseService.kt
package com.yangxj96.saas.common.base

import com.baomidou.mybatisplus.extension.service.IService

/**
 * RESTFul 接口公用service层
 *
 * @param <O> 子类对应的实体
</O> */
interface BaseService<O : BaseEntity> : IService<O> {

    fun create(datum: O): O

    fun delete(id: String): Boolean

    fun modify(datum: O): O

}
